openduck-py/openduck_py/response_agent.py

```diff
@@ -375,6 +376,11 @@ async def _generate_and_speak(
                 history_json={"messages": [system_prompt]},
             )
             db.add(chat)
+        else:
+            messages = chat.history_json["messages"]
+            if messages[0]["role"] == "system":
+                messages[0] = system_prompt
+            chat.history_json["messages"] = messages
```

Comment:
Does this if statement make any changes? It looks to me like you set it equal to the code below if the role in the messages is set to system – but that is idempotent

```python
system_prompt = {
  "role": "system",
  "content": system_prompt,
}
```

Reply:
yeah it's a bit confusing. on the first round, _generate_and_speak is called with a different system prompt: https://github.com/uberduck-ai/openduck/blob/main/openduck-py/openduck_py/routers/voice.py#L305

and then this was being used as the system prompt for all future messages, because it was stored in the DB.

Reply:
Responded... Lmk if it makes sense

Reply:
Ok got it, so this just overwrites the first system prompt with the new version?

Reply:
yep!